File: problemas/ad-hoc/2189/2189.c
#include <stdio.h>

int main(){
    int n, numero;

    int teste = 1;
    while(1) {
        scanf("%d\n", &n);

        if (n == 0) {
            break;
        }

        int resposta = 0;
        for(int i = 1; i <= n; ++i) {
            scanf("%d", &numero);
            if (i == numero) {
                resposta = numero;
            }
        }

        printf("Teste %d\n%d\n\n", teste, resposta);
        teste += 1;
    }

    return 0;
}

File: problemas/ad-hoc/2189/2189.kt
import java.util.*

fun main(args: Array<String>) {
    val sc = Scanner(System.`in`)

    var teste = 1
    while (true) {
        val n: Int = sc.next().toInt()
        if (n == 0) {
            break
        }

        var resposta = -1
        for (i in 1..n) {
            val numero: Int = sc.next().toInt()
            if (i == numero) {
                resposta = i
            }
        }

        println("Teste " + teste)
        println(resposta)
        println("")

        teste += 1
    }
}
